duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -175,299 +205,480 @@ namespace caf {
-        sr.RegCNNNueE = v[0];
+    for (unsigned int n = 0; n < particleVector.size(); ++n) {
+      const art::Ptr<recob::PFParticle> particle = particleVector.at(n);
+      if(particle->IsPrimary()){
+        continue;
       }
+
+      caf::SRRecoParticle particle_record;
+      particle_record.primary = (particle->Parent() == nuID);
+      //For now PDG is taken from the PFP only which does not include PID info beyond track/shower
+      particle_record.pdg = particle->PdgCode();
+
+      //TODO: Add the energy reco information. Requires some uniform framework and specific choice about the method to use.
+
     }
+  }
 
-    auto mct = evt.getHandle< std::vector<simb::MCTruth> >("generator");
+
+  //------------------------------------------------------------------------------
+
+  int CAFMaker::FillGENIERecord(simb::MCTruth const& mctruth, simb::GTruth const& gtruth)
+  {
+    genie::EventRecord* record = evgb::RetrieveGHEP(mctruth, gtruth);
+    int cur_idx = fGENIETree->GetEntries();
+    fEventRecord->Fill(cur_idx, record);
+    fGENIETree->Fill();
+
+    delete record;
+
+    return cur_idx;
+  }
+
+
+  //------------------------------------------------------------------------------
+  
+  void CAFMaker::analyze(art::Event const & evt)
+  {
+    caf::StandardRecord sr;
+    caf::StandardRecord* psr = &sr;
+    
+
+    if(fTree){
+      
+      fTree->SetBranchAddress("rec", &psr);
+    }
+
+    art::ServiceHandle<geo::Geometry const> fGeometry;
+    std::string geoName = fGeometry->DetectorName();
+
+    std::cout << "Geo name is: " << geoName << std::endl;
+
+    SRDetectorMeta *detector;
+
+    if(geoName.find("dunevd10kt") != std::string::npos){
+      detector = &(sr.meta.fd_vd);
+      std::cout << "Assuming the FD VD detector" << std::endl;
+    }
+    else if (geoName.find("dune10kt") != std::string::npos)
+    {
+      detector = &(sr.meta.fd_hd);
+      std::cout << "Assuming the FD HD detector" << std::endl;
+    }
+    else {
+      std::cout << "Warning: Didn't detect a know geometry. Defaulting to FD HD!" << std::endl;
+      detector = &(sr.meta.fd_hd);
+    }
+    
+
+
+
+    FillMetaInfo(*detector, evt);
+
+    FillBeamInfo(sr.beam, evt);
+    art::Handle<std::vector<simb::MCTruth>> mct = evt.getHandle< std::vector<simb::MCTruth> >("generator");
+    art::Handle<std::vector<simb::GTruth>> gt = evt.getHandle< std::vector<simb::GTruth> >("generator");
     if ( !mct ) {
-      mf::LogWarning("CAFMaker") << "No MCTruth.";
-        }
-    else if ( !mct->empty() ) {
-      FillTruthInfo(sr, *mct, evt.getProduct<std::vector<simb::MCFlux> >("generator"), evt);
-      }
+      mf::LogWarning("CAFMaker") << "No MCTruth. SRTruthBranch will be empty!";
+    }
+    else if ( !gt ) {
+      mf::LogWarning("CAFMaker") << "No GTruth. SRTruthBranch will be empty!";
+    }
+    else {
+      FillTruthInfo(sr.mc, *mct, *gt, evt.getProduct<std::vector<simb::MCFlux> >("generator"), evt);
```

Comment:
It might just be personal preference, but I wasn't a fan of having `evt.getProduct<std::vector<simb::MCFlux> >("generator")` in the function call. Especially because you check if the other products are filled.

Reply:
This was fixed.